packages/core/test/invoke.test.ts

```diff
@@ -2709,6 +2709,31 @@ describe('invoke', () => {
       interpret(machine).start();
     });
   });
+
+  it('invoke creator should provide reference to self', (done) => {
```

Comment:
Especially `getSnapshot` is going to be quite useful too. I think it does deserve an additional, basic, test.

Reply:
This is done in 4c3b9b9653a333bb347c00f715a6e0e3784f95d0